src/useIsShake.ts

```diff
@@ -51,14 +54,41 @@ export function useIsShake() {
     setSubscription(null);
   };
 
+  const checkAccelerometerAvail = async () => {
+    const availability = await isAvailableAsync();
+    setIsAccelerometerAvail(() => {
+      setIsAccelerometerAvailPending(false);
+      return availability;
+    });
+  };
+
+  const getAccelerometerPermiss = async () => {
+    const permissionResponse = await requestPermissionsAsync();
+    if (permissionResponse.granted) setIsAccelerometerAvail(() => {
+      setIsAccelerometerAvailPending(false);
+      return permissionResponse.granted;
+    }); // following else if block for when permissionResponse.canAskAgain === false in order to direct end user to Settings app in order to enable permission to access Accelerometer
+    else if (!permissionResponse.canAskAgain) {
+      Linking.openSettings();
+    };
+  };
+
   useEffect(() => {
+    if (isAccelerometerAvail === null && !isAccelerometerAvailPending) {
+      setIsAccelerometerAvailPending(true);
+      checkAccelerometerAvail();
+    };
+    if (isAccelerometerAvail === false && !isAccelerometerAvailPending) {
+      setIsAccelerometerAvailPending(true);
+      getAccelerometerPermiss();
+    };
     // invocation of _subscribe when isToggleReady === true
-    if (isShakeReady) {
+    if (isAccelerometerAvail && isShakeReady) {
```

Comment:
Adding listener to accelerometer is now dependent on the accelerometer being available to app.